src/themeProvider.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, FC, useContext, useState } from 'react'
+import { ThemeProvider as StyledThemeProvider, useTheme } from 'styled-components/native'
+type ColorMode = 'light' | 'dark'
+
+export type ThemeSetters = {
+  setMode: (mode: ColorMode) => void
+  setFontScale: (scale: number) => void
+}
+
+export type Theme = {
+  mode: ColorMode
+  fontScale: number
+}
+
+const ThemeSetterContext = createContext<ThemeSetters>({} as any)
+
+/**
+ * Global Theme Provider.
+ */
+const ThemeProvider: FC = ({ children }) => {
+  const [mode, setMode] = useState<ColorMode>('light')
+  const [fontScale, setFontScale] = useState<number>(1)
```

Comment:
I'm skeptical about introducing a completely new method of state management. **em** is a Redux application and all global state is in the store. Access to state is provided through standard actions and reducers. This provider presents a new approach to storing global state and a new approach to modifying it, via hooks. This would be fine for a new application, but I'm not sure it's justified in an existing application with well-established state management norms.

Reply:
Sure we can push this into the redux state. I did this for a quick demonstration of `themeSpeific` helper.

Reply:
Okay, I see. 

In the future, please make explicit which aspects are for demonstration purposes only. I run into this a lot with draft PR's. Neither of us need me to explain things that you already know, but without explicit comments, there is no way for me to infer your intentions about which parts of the code are meant to be evaluated.